libraries/state_history/include/eosio/state_history/rocksdb_receiver.hpp

```diff
@@ -0,0 +1,121 @@
+ *   does not contain proper values for payer so it can't be used to create its delta) associated with them and
+ *   the second call for the object itself. For that reason there is logic also associated to detect when we want
+ *   to store the table_id_object_view itself (which is when we receive only 1 call for this and the next call is not
+ *   any of the primary or secondary indexes).
+ *
+ *   2. When operating on the not sigle entry mode, then we store the deltas of the objects as we receive them.
+ *
+ */
+class rocksdb_receiver {
+public:
+   rocksdb_receiver(std::vector<table_delta> &deltas, const chainbase::database& db, bool single_entry = false) : deltas(deltas), db_(db), single_entry(single_entry) {};
+
+   void add_row(const chain::kv_object_view& row) {
+      if(single_entry) {
+         maybe_process_table();
+      }
+
+      deltas[get_delta_index("key_value")].rows.obj.emplace_back(present, fc::raw::pack(make_history_serial_wrapper(db_, row)));
+      table_ = {};
+   }
+
+   void add_row(const chain::backing_store::table_id_object_view& row) {
+      if(single_entry) {
+         maybe_process_table();
+      } else {
+         deltas[get_delta_index("contract_table")].rows.obj.emplace_back(present, fc::raw::pack(make_history_serial_wrapper(db_, row)));
+      }
+
+      table_ = row;
+   }
+
+   void add_row(const chain::backing_store::primary_index_view& row) {
+      deltas[get_delta_index("contract_row")].rows.obj.emplace_back(present, fc::raw::pack(make_history_context_wrapper(db_, *table_, row)));
+      table_ = {};
+   }
+
+   void add_row(const chain::backing_store::secondary_index_view<uint64_t>& row) {
+      deltas[get_delta_index("contract_index64")].rows.obj.emplace_back(present, fc::raw::pack(make_history_context_wrapper(db_, *table_, row)));
+      table_ = {};
+   }
+
+   void add_row(const chain::backing_store::secondary_index_view<chain::uint128_t>& row) {
+      deltas[get_delta_index("contract_index128")].rows.obj.emplace_back(present, fc::raw::pack(make_history_context_wrapper(db_, *table_, row)));
+      table_ = {};
+   }
+
+   void add_row(const chain::backing_store::secondary_index_view<chain::key256_t>& row) {
+      deltas[get_delta_index("contract_index256")].rows.obj.emplace_back(present, fc::raw::pack(make_history_context_wrapper(db_, *table_, row)));
+      table_ = {};
+   }
+
+   void add_row(const chain::backing_store::secondary_index_view<float64_t>& row) {
+      deltas[get_delta_index("contract_index_double")].rows.obj.emplace_back(present, fc::raw::pack(make_history_context_wrapper(db_, *table_, row)));
+      table_ = {};
+   }
+
+   void add_row(const chain::backing_store::secondary_index_view<float128_t>& row) {
+      deltas[get_delta_index("contract_index_long_double")].rows.obj.emplace_back(present, fc::raw::pack(make_history_context_wrapper(db_, *table_, row)));
+      table_ = {};
+   }
+
+   void add_row(fc::unsigned_int x) {
```

Comment:
Is this needed?

Reply:
yes, the method needs to be there to be called, but this use case doesn't care about it

Reply:
I added a comment explaining the reason.